src/Fixer/PhpUnit/PhpUnitNamespacedFixer.php

```diff
@@ -0,0 +1,167 @@
+{
+    /**
+     * @var string
+     */
+    private $originalClassRegEx;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'PHPUnit classes MUST be used in namespaced version, eg `\PHPUnit\Framework\TestCase` instead of `\PHPUnit_Framework_TestCase`.',
+            [
+                new CodeSample(
+'<?php
+final class MyTest extends \PHPUnit_Framework_TestCase
+{
+}
+'
+                ),
+            ],
+            "PHPUnit v6 has finally fully switched to namespaces.\n"
+            ."You could start preparing the upgrade by switching from non-namespaced TestCase to namespaced one.\n"
+            .'Forward compatibility layer (`\PHPUnit\Framework\TestCase` class) was backported to PHPUnit v4.8.35 and PHPUnit v5.4.0.'."\n"
+            .'Extended forward compatibility layer (`PHPUnit\Framework\Assert`, `PHPUnit\Framework\BaseTestListener`, `PHPUnit\Framework\TestListener` classes) was introduced in v5.7.0.'."\n",
+            'Risky when PHPUnit classes are overridden or not accessible, or when project has PHPUnit incompatibilities.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isTokenKindFound(T_CLASS);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isRisky()
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        if (PhpUnitTargetVersion::fulfills($this->configuration['target'], PhpUnitTargetVersion::VERSION_6_0)) {
+            $this->originalClassRegEx = '/^PHPUnit_\w+$/i';
+        } elseif (PhpUnitTargetVersion::fulfills($this->configuration['target'], PhpUnitTargetVersion::VERSION_5_7)) {
+            $this->originalClassRegEx = '/^PHPUnit_Framework_TestCase|PHPUnit_Framework_Assert|PHPUnit_Framework_BaseTestListener|PHPUnit_Framework_TestListener$/i';
+        } else {
+            $this->originalClassRegEx = '/^PHPUnit_Framework_TestCase$/i';
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $importedOriginalClassesMap = [];
+        $currIndex = 0;
+
+        while (null !== $currIndex) {
+            $match = $tokens->findSequence([[T_STRING]], $currIndex);
+
+            if (null === $match) {
+                break;
+            }
+
+            $matchIndexes = array_keys($match);
+            $currIndex = $matchIndexes[0];
+
+            $originalClass = $match[$currIndex]->getContent();
+
+            if (1 !== preg_match($this->originalClassRegEx, $originalClass)) {
+                ++$currIndex;
+
+                continue;
+            }
+
+            $substituteTokens = $this->generateReplacement($originalClass);
+
+            $tokens->clearAt($currIndex);
```

Comment:
do you need the clear at `index` if you insert at the same `index` after that?

Reply:
I need it, as inserting new token is not removing old token. Also, I do not replace one token with another.
I replace one token with bunch of tokens, so I remove old one and insert new ones.

Reply:
ah yes, we deprecated the `override` methods, so all good :+1: